am-lib/src/main/java/uk/gov/hmcts/reform/amlib/DefaultRoleSetupImportService.java

```diff
@@ -0,0 +1,130 @@
+package uk.gov.hmcts.reform.amlib;
+
+import lombok.NonNull;
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.sqlobject.SqlObjectPlugin;
+import uk.gov.hmcts.reform.amlib.enums.AccessManagementType;
+import uk.gov.hmcts.reform.amlib.enums.RoleType;
+import uk.gov.hmcts.reform.amlib.enums.SecurityClassification;
+import uk.gov.hmcts.reform.amlib.exceptions.ErrorAddingEntriesToDatabaseException;
+import uk.gov.hmcts.reform.amlib.models.DefaultPermission;
+import uk.gov.hmcts.reform.amlib.models.DefaultPermissionGrant;
+import uk.gov.hmcts.reform.amlib.models.ResourceAttribute;
+import uk.gov.hmcts.reform.amlib.repositories.DefaultRoleSetupRepository;
+import uk.gov.hmcts.reform.amlib.utils.Permissions;
+
+public class DefaultRoleSetupImportService {
+    private final Jdbi jdbi;
+
+    public DefaultRoleSetupImportService(String url, String user, String password) {
```

Comment:
I have a suspicious that passing `DataSource` makes more sense because then library consumers can pass data source that implements connection pool.

Can you add alternate constructor that accepts solely `DataSource`?

Reply:
The thing is that when you pass an object to constructor user of lib must also import this class. 

If you pass strings it's more independent

Reply:
`DataSource` is one of the standard JDBC classes so import is not a problem. Using data source massively change JDBI works as it does not need to make new connection to DB for every query. When data source is used (especially one backed by connection poll) it largely improves performance and removes problem of running out of DB connections.

Reply:
@swalker125 I am bit worried that exposing string based constructor will be error prone. I suggest either removing or maybe describing consequences in Javadoc.